src/platform/plugins/shared/workflows_management/public/widgets/workflow_yaml_editor/lib/snippets/generate_connector_snippet.ts

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { isMac } from '../../../../shared/utils/is_mac';
+import { getRequiredParamsForConnector } from '../get_required_params_for_connector';
+
+interface GenerateConnectorSnippetOptions {
+  full?: boolean;
+}
+
+/**
+ * Generates a YAML snippet for a workflow connector step based on the specified type.
+ * @param connectorType - The type of connector to generate a snippet for
+ * @param options - Configuration options for snippet generation
+ * @param options.full - Whether to include the full YAML structure with step name and type prefix
+ * @returns The formatted YAML connector snippet with required parameters as placeholders
+ */
+export function generateConnectorSnippet(
+  connectorType: string,
+  { full }: GenerateConnectorSnippetOptions = {}
+): string {
+  let prepend = '';
+  if (full) {
+    prepend = `- name: ${connectorType}_step\n  type: `;
+  }
+
+  // Get required parameters for this connector type
+  const requiredParams = getRequiredParamsForConnector(connectorType);
+
+  if (requiredParams.length === 0) {
```

Comment:
Just wondering if we can define template strings with nunjucks syntax and then build these strings from templates